mongo_connector/oplog_manager.py

```diff
@@ -183,12 +196,42 @@ def oplog_ns_set(self):
     def update_oplog_ns_set(self):
         self._oplog_ns_set = []
         if self.namespace_set:
-            self._oplog_ns_set.extend(self.namespace_set)
+            for ns in self.namespace_set:
+                if "*" in ns:
+                    # if namespace_set contains *, change to regexp
+                    reg_ns = re.sub(r"(\.)","\.", ns)
+                    reg_ns = re.sub(r"(\*)", ".*", reg_ns)
+                    LOG.debug("OplogThread: Replace namespace %s to %s"  %(ns, reg_ns))
+                    reg = re.compile(reg_ns)
+                    self._oplog_ns_set.append(reg)
+                else:
+                    self._oplog_ns_set.append(ns)
+                    self._oplog_ns_set.append(ns.split('.', 1)[0] + '.$cmd')
+
             self._oplog_ns_set.extend(self.gridfs_files_set)
-            self._oplog_ns_set.extend(set(
-                ns.split('.', 1)[0] + '.$cmd' for ns in self.namespace_set))
             self._oplog_ns_set.append("admin.$cmd")
 
+    @property
+    def oplog_ex_ns_set(self):
+        try:
+            return self._oplog_ex_ns_set
+        except AttributeError:
```

Comment:
Ooops - I meant to say that we call [`self.ex_namespace_set = ...`](https://github.com/mongodb-labs/mongo-connector/pull/495/files/19ab0ed1658d184061f3312994cb811a04c745a9#diff-9e5fc676a1f3b77e19871877266b4cf0R79) in the constructor, which calls `update_oplog_ex_ns_set`, which in turn sets self._oplog_ex_ns_set.

I don't know why the other getter methods are using try/except AttributeError, since those seem similarly impossible. I'd say remove the try/except here. I'll run the tests again to determine if the try/except cases are necessary in the other getters, though I can't see why they would be.